tests/runtime/conftest.py

```diff
@@ -149,6 +161,47 @@ def runtime_grpc_server(
         yield server
 
 
+@contextmanager
+def runtime_http_test_server(open_port, *args, **kwargs):
+    """Helper to wrap creation of RuntimeHTTPServer in temporary configurations"""
+    with tempfile.TemporaryDirectory() as workdir:
+        temp_log_dir = os.path.join(workdir, "metering_logs")
+        temp_save_dir = os.path.join(workdir, "training_output")
+        os.makedirs(temp_log_dir)
+        os.makedirs(temp_save_dir)
+        with temp_config(
+            {
+                "runtime": {
+                    "metering": {"log_dir": temp_log_dir},
+                    "training": {"output_dir": temp_save_dir},
+                    "http": {"port": open_port},
+                }
+            },
+            "merge",
+        ):
+            with http_server.RuntimeHTTPServer(*args, **kwargs) as server:
+                # Give tests access to the workdir
+                server.workdir = workdir
```

Comment:
I think we should throw `_check_http_server_readiness(server)` in here before the `yield`

Reply:
That said, we'd need to update `_check_http_server_readiness` to support TLS too